### ISSUE TITLE
BottomAppBar: `surfaceTintColor` doc wrongly says "If this is null, no overlay will be applied."

### ISSUE THREAD

**Issue by chrisbobbe** [flutter/flutter#124535]
Here's the full doc on [`BottomAppBar.surfaceTintColor`](https://api.flutter.dev/flutter/material/BottomAppBar/surfaceTintColor.html) currently:

```dart
  /// The color used as an overlay on [color] to indicate elevation.
  ///
  /// If this is null, no overlay will be applied. Otherwise the
  /// color will be composited on top of [color] with an opacity related
  /// to [elevation] and used to paint the background of the [BottomAppBar].
  ///
  /// The default is null.
  ///
  /// See [Material.surfaceTintColor] for more details on how this overlay is applied.
  final Color? surfaceTintColor;
```

But when `surfaceTintColor` is `null`, an overlay _is_ applied. In this code sample, compare a `Container` with a `BottomAppBar` that are both given the same gray for their `color` param:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    Color color = const Color.fromRGBO(90, 90, 90, 1);
    return MaterialApp(
      theme: Theme.of(context).copyWith(useMaterial3: true),
      home: Scaffold(
        body: Align(alignment: Alignment.bottomCenter, child: Container(width: 100, height: 100, color: color)),
        bottomNavigationBar: BottomAppBar(
          color: color,
          child: Row(children: [IconButton(onPressed: () {}, icon: const Icon(Icons.menu))])
        ),
      ),
    );
  }
}
```

<img width=400 src="https://user-images.githubusercontent.com/22248748/230963143-fbfecea4-9ee4-4065-8804-4e52f9395ece.png">

The bottom app bar gets its bluish tint from a default surface tint color, which I think in this case is `ColorScheme.primary`: https://github.com/flutter/flutter/blob/bf0bdacb295697ebcac88cc6b1062cdb2f6d6552/packages/flutter/lib/src/material/color_scheme.dart#L727

Probably the behavior is right, and the doc should be updated to say something like:

```dart
  /// A custom color for the Material 3 surface-tint elevation effect.
  ///
  /// In Material 3, a "surface tint" with an opacity related to [elevation]
  /// will be applied to the [BottomAppBar]'s background.
  /// Use this property to override the default color of that tint.
  ///
  /// If this property is null, then [BottomAppBarTheme.surfaceTintColor]
  /// of [ThemeData.bottomAppBarTheme] is used.
  /// If that is also null, [ColorScheme.surfaceTint] is used.
  ///
  /// Ignored if [ThemeData.useMaterial3] is false.
  ///
  /// The default is null.
  ///
  /// See [Material.surfaceTintColor] for more details on how this overlay is applied.
  final Color? surfaceTintColor;
```

**Comment by chrisbobbe** [comment]
See also #124536, in which I propose the same draft doc for this property.

**Comment by rydmike** [comment]
@chrisbobbe, both mentioned comments are valid points. I had noticed this earlier as well, but I'm also so used to tint just applying in M3 mode that I did not think much of it.

It is is of course a documentation update oversight. I think there are potentially many places where same clarification applies.

One cause of this documentation oversight might also be that originally, in earlier stages of M3 support being added, the underlying used `Material` did indeed work as described in these comments, but it was later changed. The forgotten doc updates are most likely due to that change in behavior.
